sim/src/main/scala/spinal/sim/VerilatorBackend.scala

```diff
@@ -491,6 +558,16 @@ JNIEXPORT void API JNICALL ${jniPrefix}disableWave_1${uniqueId}
     assert(s"make -j$threadCount VM_PARALLEL_BUILDS=1 -C ${workspacePath}/${workspaceName} -f V${config.toplevelName}.mk V${config.toplevelName} CURDIR=${workspacePath}/${workspaceName}".!  (new Logger()) == 0, "Verilator C++ model compilation failed")
 
     FileUtils.copyFile(new File(s"${workspacePath}/${workspaceName}/V${config.toplevelName}${if(isWindows) ".exe" else ""}") , new File(s"${workspacePath}/${workspaceName}/${workspaceName}_$uniqueId.${if(isWindows) "dll" else (if(isMac) "dylib" else "so")}"))
+
+    if (cacheEnabled) {
+      // update cache
+
+      val workspaceDir = new File(s"${workspacePath}/${workspaceName}")
+      val workspaceCacheDir = new File(s"${cachePath}/${workspaceName}")
+
+      FileUtils.deleteQuietly(workspaceCacheDir)
+      FileUtils.copyDirectory(workspaceDir, workspaceCacheDir)
```

Comment:
If only the `*.so`/`*.dylib`/`*.dll` file is needed (?), I could copy that file instead of the whole directory to save disk space. Then I could also save multiple versions of this file without creating a huge cache (think of multiple simulation configs, always switching between them).

Reply:
I'm not exactly sure, but i would say : 
*.so/*.dylib/*.dll is enough. You don't need to include the bin / mem, because they are embeded into the verilator model.

Reply:
Just tested it. You only have to copy the `*.so`/`*.dylib`/`*.dll`. I also realized that the shared library contains the `uniqueId` which must match also match the assigned `uniqueId` when using the cached binary. So I now changed the code to only copy the `__ALL.a`, `*.mk` and `*.h` files and also always (re)run make but skip the verilator compilation when the cache is used. So the wrapper will always be compiled with the assigned `uniqueId`.